[rb] Remove `prism` dependency

Earlopain opened the pull request:
### **User description**
This project does not use prism at all. At most it is being used by rubocop during local development.

It was added in https://github.com/SeleniumHQ/selenium/pull/16357 but it is unclear to me why. I just bumped it for this repo, perhaps that JRuby issue has since been fixed. Either way, it should not be part of the gemspec.

<!-- Thanks for Contributing to Selenium! -->
<!-- Please read our contribution guidelines: https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md -->

### 🔗 Related Issues
Added in https://github.com/SeleniumHQ/selenium/pull/16357

### 💥 What does this PR do?
Remove prism as a dependency and bump it.

### 🔧 Implementation Notes
selenium-webdriver does not use prism at all


### 🔄 Types of changes
<!-- ✂️ Please delete anything that doesn't apply -->
- Bug fix (backwards compatible)


___

### **PR Type**
Bug fix


___

### **Description**
- Remove unused `prism` dependency from gemspec

- Bump `prism` version in MODULE.bazel to 1.5.2

- Clean up unnecessary runtime dependency


___

### Diagram Walkthrough


```mermaid
flowchart LR
  gemspec["selenium-webdriver.gemspec"] -- "remove dependency" --> prism["prism gem"]
  bazel["MODULE.bazel"] -- "bump version" --> prism_version["prism 1.5.2"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>selenium-webdriver.gemspec</strong><dd><code>Remove prism dependency from gemspec</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rb/selenium-webdriver.gemspec

<ul><li>Removed <code>prism</code> dependency from gemspec dependencies list<br> <li> Dependency was unused by the project at runtime</ul>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16437/files#diff-f7c3979ae889cfb6e1b2386569a8bcd4f45e9a7e03d8c13bad52eb502a240aad">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>MODULE.bazel</strong><dd><code>Bump prism version to 1.5.2</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

MODULE.bazel

<ul><li>Updated <code>prism</code> version from 1.4.0 to 1.5.2<br> <li> Bumped SHA hash for new version</ul>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16437/files#diff-6136fc12446089c3db7360e923203dd114b6a1466252e71667c6791c20fe6bdc">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

Earlopain committed:
[rb] Remove `prism` dependency

This project does not use prism at all. At most it is being used by rubocop during local development.

It was added in https://github.com/SeleniumHQ/selenium/pull/16357 but it is unclear to me why.
I just bumped it for this repo, perhaps that JRuby issue has since been fixed.
Either way, it should not be part of the gemspec.

Earlopain commented:
If there is something broken with prism on jruby, please let me know. I am a contributor to `prism`.

diemol commented:
I think there was a conflict with Rubocop. Let's run the CI and see if it fails again.

diemol committed:
Merge branch 'trunk' into rb-remove-prism-dep

Earlopain committed:
Pin prism to 1.4.0 for the project

It fails to build with the following error:
src/prism.c: In function ‘context_terminator’:
src/prism.c:8651:62: error: conversion to ‘unsigned int’ from ‘int’ may change
the sign of the result [-Werror=sign-conversion]
8651 |     return token->type < 32 && (context_terminators[context] & (1 <<
token->type));
      |                                                              ^
cc1: all warnings being treated as errors

Earlopain commented:
Ok, I see the error. It's a build failure that I don't think is related to JRuby at all. I have pinned back to 1.4.0 in the Gemfile to allow the build to pass. Specifying it there allows consumers of `selenium-webdriver` to update their own copy of prism.

I will see about getting that warning fixed in prism so that it can be unpinned here. Unsure why it fails your build here, I guess some compiler option that ruby itself doesn't test against.

diemol committed:
Merge branch 'trunk' into rb-remove-prism-dep

diemol reviewed (approved):
Thank you!

Earlopain commented:
The build failure may be fixed in the next prism release: https://github.com/ruby/prism/pull/3682

boris-petrov commented:
@Earlopain I'm not sure what specific JRuby issue they are talking about here but I have [this](https://github.com/jruby/jruby/issues/8727) specific JRuby issue with `prism`. There is a reproduction [repo](https://github.com/jruby/jruby/issues/8727#issuecomment-3075931510). Note, however, that this is most likely a JRuby issue, not a `prism` one, but just posting this here as you might have any idea...
